nhs/data/join_on_sa1.py

```diff
@@ -0,0 +1,219 @@
+from typing import Any, List  # Importing required types
+
+import fiona
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import numpy as np
+import polars as pl
+import yaml
+from fiona.drvsupport import supported_drivers
+from loguru import logger
+from pathos.multiprocessing import ProcessingPool as Pool
+from pyproj import CRS
+
+# Set the SHAPE_RESTORE_SHX option
+supported_drivers["ESRI Shapefile"] = "rw"
+
+
+def read_config(config_path: str) -> dict:
+    """Read YAML configuration file."""
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+    logger.info(f"Configuration loaded: {config}")
+    return config
+
+
+def read_shapefile(path: str) -> int:
+    """Read shapefile and return total number of rows."""
+    total_rows = len(pl.read_parquet(path))
+    logger.info(f"Total number of rows in the shapefile: {total_rows}")
+    return total_rows
```

Comment:
Why is reading the shapefile function, reading a parquet?

Reply:
Ah good catch, there is apparently a feature to geoparquet these types of files but I think it's paywalled so I'll keep the shapefile as being un-parqueted.